File: Backend/src/modules/instance/dto/instance_viewer_meta.rs
use crate::modules::instance::dto::InstanceViewerGuild;

#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
pub struct InstanceViewerMeta {
    pub instance_meta_id: u32,
    pub guild: Option<InstanceViewerGuild>,
    pub server_id: u32,
    pub expansion_id: u8,
    pub map_id: u16,
    pub map_difficulty: Option<u8>,
    pub start_ts: u64,
    pub end_ts: Option<u64>,
    pub expired: Option<u64>,
    pub upload_id: u32,
}

File: Webclient/src/app/module/viewer/domain_value/instance_viewer_meta.ts
import {InstanceViewerGuild} from "./instance_viewer_guild";

export interface InstanceViewerMeta {
    instance_meta_id: number;
    guild: InstanceViewerGuild | null;
    server_id: number;
    expansion_id: number;
    map_id: number;
    map_difficulty: number | null;
    start_ts: number;
    end_ts: number | null;
    expired: number | null;
    upload_id: number;
}
